react-payments-app/src/components/domain/Card/Card.jsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { useCard } from '../../../store/CardContext';
+import { cardCompanies } from '../../../server/cardCompanies';
+
+const Card = ({
+  cardName,
+  cardNumbers,
+  cardOwner,
+  cardExpirationDate,
+  cardNickname,
+  onClick,
+}) => {
+  const { cardInfo } = useCard();
+  const [cardUI, setCardUI] = useState('empty-card');
+  const location = useLocation();
+  const currentLocation = location.pathname.split('/').slice(-1).shift();
+
+  //TODO: 회사 선택 유무에 따라 className 변경
+  // TODO: 회사에 따라 className 변경
+
+  useEffect(() => {
+    const id = cardInfo.cardCompanyId;
+
+    if (id) {
+      setCardUI('small-card');
+    }
+    if (currentLocation === 'completed') {
+      setCardUI('big-card');
+    }
+  }, []);
+
+  return (
+    <>
+      <div id='card' onClick={onClick}>
+        <div className='flex-column-center'>
+          <div className='card-box'>
+            <div className={cardUI}>
```

Comment:
cardUI가 꼭 상태여야할 이유가 있을까요?


```suggestion
  const location = useLocation();
  const currentLocation = location.pathname.split('/').slice(-1).shift();

  return (
    <>
      <div id='card' onClick={onClick}>
        <div className='flex-column-center'>
          <div className='card-box'>
            <div className={ getCardUI({id, currentLocation}) }>
```

이런식으로 하는건 어떨까요?

Reply:
실시간으로 변경되는 값일 필요 없으니 메소드로 하는 편이 깔끔하겠다는 의도이실까요?

Reply:
[455a6c2](https://github.com/next-step/react-payments/pull/84/commits/455a6c29321e1c2ad93919aedf35d8cf9c372a0c) 에서 수정해보았습니다!

Reply:
> 실시간으로 변경되는 값일 필요 없으니 메소드로 하는 편이 깔끔하겠다는 의도이실까요?

cardUI는 직접적인 변경은 없고  id와 currentLocation에 의해서 파생되는 값이므로, 상태로 관리할 필요가 없다는 의미였어요!